lib/ansible/modules/windows/win_disk_facts.ps1

```diff
@@ -244,5 +244,8 @@ foreach ($disk in $disks) {
     $result.ansible_facts.ansible_disks += $disk_info
 }
 
+# Sort by disk number property
+$result.ansible_facts.ansible_disks = @() + ($result.ansible_facts.ansible_disks | Sort-Object -Property Number)
```

Comment:
`ansible_disks` is a hashtable so `-Property Name` does not work normally. What you need to do is sort by a property expression like `| Sort-Object -Property @{e={$_.Number}}`.

Reply:
But ansible_disks is an array. From tests on my machine it worked for multiple disks as well. You mean on lower versions maybe?

Reply:
It's an array of hashtables and we are trying to sort it by the "Number" property. Because each element in the array is a hashtable it doesn't actually have a property called `Number` but rather a key called `number`. Typically accessing this key is the same as getting an object's property, i.e. `$hashtable.number`. This is because PowerShell hides these differences in the background but this can affect things like `Sort-Object`. As an example here is where it can go wrong

```
$data = @(
    @{name='a'}
    @{name='c'}
    @{name='e'}
    @{name='f'}
    @{name='d'}
    @{name='b'}
)
$data | Sort-Object -Property name
```

Running that on PS v5.1 I get

```
PS C:\Users\vagrant-domain> $data | Sort-Object -Property name

Name                           Value
----                           -----
name                           f
name                           d
name                           b
name                           a
name                           c
name                           e
```

Instead we should just be providing a way for `Sort-Object` to get the value to sort by, e.g. `-Property {$_.name}` which then becomes

```
PS C:\Users\vagrant-domain> $data | Sort-Object -Property {$_.name}

Name                           Value
----                           -----
name                           a
name                           b
name                           c
name                           d
name                           e
name                           f
```

It seems like this may or may not be the case for PowerShell Core but we have to worry about Windows PowerShell.